README.md

```diff
@@ -121,15 +118,23 @@ One of the basic features of an Algorand SDK is the ability to construct all typ
 
 ### Encoding Utilities
 
-In order to ensure transactions are compact and can hash consistently, there are some special encoding requirements. The SDKs must provide utilities to work with these encodings. Algorand uses MessagePack as a compact binary-encoded JSON alternative, and fields with default values are excluded from the encoded object. Additionally to ensure consistent hashes, the fields must be alphebatized.
+To ensure that transactions are compact and can hash consistently, there are some special encoding requirements. The SDKs must provide utilities to work with these encodings. Algorand uses MessagePack as a compact binary-encoded JSON alternative, and fields with default values are excluded from the encoded object. Additionally to ensure consistent hashes, the fields must be alphabetized. Finally, dApp args and return values can be encoded and decoded in accordance with the [ARC-004 ABI-Type Specification](https://arc.algorand.foundation/ARCs/arc-0004#encoding).
 
 ### Crypto Utilities
 
 All things related to crypto to make it easier for developers to work with the blockchain. This includes standard things like ED25519 signing, up through Algorand specific LogicSig and MultiSig utilities. There are also some convenience methods for converting Mnemonics.
 
-### TEAL Utilities
+### Enriched Interaction
 
-Everything related to working with [TEAL](https://developer.algorand.org/docs/reference/teal/specification/#transaction-execution-approval-language-teal). This includes some utilities for parsing and validating compiled TEAL programs.
```

Comment:
do we not have utilities for parsing and validating compiled TEAL programs anymore?

Reply:
I didn't find them in Python. I'll search a bit more in other SDK's and in the history.